torchreid/engine/image/am_softmax.py

```diff
@@ -206,13 +206,13 @@ def forward_backward(self, data):
 
             total_loss += mutual_loss / float(num_mutual_losses)
 
-        total_loss.backward(retain_graph=self.enable_metric_losses)
+        total_loss.backward()
```

Comment:
I'm not sure that it will work without `retain_graph` (for ml losses) - as far as I know each call of `step()` method clears grads and any future call of `step()` will fail. Maybe I'm wrong.

Reply:
According to the documentation of pytorch 1.7 the step method doesn't affect .grad fields
```
Unless otherwise specified, this function should not modify the .grad field of the parameters.
```
https://pytorch.org/docs/stable/optim.html#torch.optim.Optimizer.step

Reply:
Originally retain_graph was enabled to allow the second backward.

Reply:
Did you have a chance to run it just for single iteration?

Reply:
Yes, I performed 500 iterations on imagenet with the center and center push losses enabled

Reply:
Ok